components/import/basic.tsx

```diff
@@ -69,32 +74,44 @@ function Basic() {
       }}
       onSubmit={async values => {
         setIsLoading(true);
-        const contract = await state.connection.contract.at(
-          values.walletAddress,
-          tzip16
-        );
-        const storage: any = await contract.storage();
-        let version = await fetchVersion(contract!);
+        setError(undefined);
 
-        let balance = await state?.connection.tz.getBalance(
-          values.walletAddress
-        );
-        let v = toStorage(version, storage, balance);
-        const validators = signers(v).map((x: string) => ({
-          address: x,
-          name: state.aliases[x] || "",
-        }));
+        try {
+          const contract = await state.connection.contract.at(
+            values.walletAddress,
+            tzip16
+          );
+          const storage: any = await contract.storage();
+          let version = await fetchVersion(contract!);
 
-        const data = {
-          ...formState,
-          ...values,
-          validators,
-          requiredSignatures: storage.threshold.toNumber(),
-          effectivePeriod: storage.effective_period.toNumber(),
-        };
-        setFormState(data as any);
-        setActiveStepIndex(activeStepIndex + 1);
-        setIsLoading(false);
+          if (version === "unknown version") {
+            throw new Error("The contract is not a TzSafe contract");
```

Comment:
this error wasn't handled on UI.
import `KT1BMLhaeartDoLXjiQZorD3yKQj6SoH6c5R` to reproduce

Reply:
Hum are you sure it's the right contract? I imported it and I got the error saying: `The contract is not a TzSafe contract`

Reply:
It shows up on UI now :thinking: (It was nothing there)